.github/workflows/macos.yml

```
@@ -39,9 +51,56 @@ jobs:
           - test_task: test-bundled-gems
             os: ${{ github.repository == 'ruby/ruby' && 'macos-arm-oss' || 'macos-14' }}
       fail-fast: false
+    runs-on: ${{ matrix.os }}
 
-    env:
-      GITPULLOPTIONS: --no-tags origin ${{ github.ref }}
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          sparse-checkout-cone-mode: false
+          sparse-checkout: /.github
+        if: ${{ matrix.test_task == 'check' || matrix.test_task == 'test-all' }}
+      - uses: ./.github/actions/setup/directories
+        with:
+          srcdir: src
+          builddir: build
+          makeup: true
+          dummy-files: ${{ matrix.test_task == 'check' }}
+          # Set fetch-depth: 10 so that Launchable can receive commits information.
+          fetch-depth: 10
+        if: ${{ matrix.test_task == 'check' || matrix.test_task == 'test-all' }}
+      - name: Set up Launchable
+        id: enable-launchable
+        uses: ./.github/actions/launchable/setup
+        with:
+          launchable-token: ${{ secrets.LAUNCHABLE_TOKEN }}
+          srcdir: src
+        if: ${{ matrix.test_task == 'check' || matrix.test_task == 'test-all' }}
+      - run: |
+          ruby -e 'File.write("launchable-subset-input.txt", Dir["test/**/*_test.rb", "test/**/test_*.rb"].map {|file| "file=#{file}"}.join("\n"))'
```

`find` command is simpler for a simple task like this.

```suggestion
          find test -name "*_test.rb" -o -name "test_*.rb" -printf "file=%P\n" > launchable-subset-input.txt
```